File: GrowthWare_VB/GrowthWareFramework/ModelObjects/ExtendedAccountProfile.vb
Namespace ModelObjects
    ''' <summary>
    ''' Example of using inheritence to extend the base AccountProfile
    ''' </summary>
    ''' <remarks></remarks>
    Public Class ExtendedAccountProfile
        Inherits MAccountProfile

#Region "Private Members"
        Private m_ExtraProperty As String = String.Empty
#End Region

#Region "Public Methods"
        Public Sub New()

        End Sub

        Public Sub New(ByVal dr As DataRow)
            On Error Resume Next
            MyBase.Init(dr)
            m_ExtraProperty = dr("ExtraProperty")
        End Sub
#End Region

#Region "Public Properties"
        Public Property extraProperty() As String
            Get
                Return m_ExtraProperty
            End Get
            Set(ByVal value As String)
                m_ExtraProperty = value.Trim
            End Set
        End Property
#End Region
    End Class
End Namespace

File: GrowthWare_CS/GrowthWareFrameworkModel/Profiles/ExtendedAccountProfile.cs

using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Web;
namespace GrowthWare.Framework.ModelObjects
{
	/// <summary>
	/// Example of using inheritence to extend the base AccountProfile
	/// </summary>
	/// <remarks></remarks>
	[Serializable(), CLSCompliant(true)]
	public class ExtendedAccountProfile : MAccountProfile
	{

		#region "Private Members"
			#endregion
		private string m_ExtraProperty = string.Empty;

		#region "Public Methods"

		public ExtendedAccountProfile()
		{
		}

		public ExtendedAccountProfile(DataRow dr)
		{
			base.init(dr);
			base.setString(ref m_ExtraProperty, ref dr, "ExtraProperty");
		}
		#endregion

		#region "Public Properties"
		public string extraProperty {
			get { return m_ExtraProperty; }
			set { m_ExtraProperty = value.Trim(); }
		}
		#endregion
	}
}
